src/common/hooks/useOnClickOutside/useOnClickOutside.ts

```diff
@@ -0,0 +1,18 @@
+import { RefObject, useEffect } from 'react';
+
+export const useOnClickOutside = (ref: RefObject<HTMLElement>, handler: (event: MouseEvent) => void) => {
+  useEffect(() => {
+    const listener = (event: MouseEvent) => {
```

Comment:
```suggestion
 const listener = (event: MouseEvent | TouchEvent) => {
```

By adding `TouchEvent` to the types of events the handler function and the listener can accept `((event: MouseEvent | TouchEvent) => void)`, you've extended the hook's utility to also work correctly on touch devices. This is crucial for maintaining usability and interactivity on mobile devices, where touch input is the primary mode of interaction.